src/features/navbar/mobile/NavbarMobile.tsx

```diff
@@ -0,0 +1,56 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+import { Menu } from 'lucide-react';
+import { NavbarMobileProps } from './model';
+import { NavItem } from '@/features';
+import { cn } from '@/lib/utils';
+
+export function NavbarMobile(props: NavbarMobileProps) {
+  const { navigation } = props;
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const menuStyles = isMenuOpen ? 'h-screen z-10' : 'h-[0rem]';
+
+  function handleOpenMenu() {
+    setIsMenuOpen(!isMenuOpen);
+  }
+
+  document.body.style.overflow = isMenuOpen ? 'hidden' : 'auto';
+  useEffect(() => {
+    const htmlEl = document.getElementById('html');
+    const bodyEl = document.getElementById('body');
+    if (!htmlEl) return;
+    if (!bodyEl) return;
+    if (isMenuOpen) {
+      htmlEl.classList.add('overflow-hidden');
+      bodyEl.classList.add('overflow-hidden');
+    } else {
+      htmlEl.classList.remove('overflow-hidden');
+      bodyEl.classList.remove('overflow-hidden');
+    }
+  }, [isMenuOpen]);
+
+  return (
+    <nav className="sticky top-0 flex justify-between bg-primary px-8 py-4 md:hidden">
+      <button type="button" onClick={handleOpenMenu}>
+        <Menu className="cursor-pointer md:hidden" />
+      </button>
+      <div
+        className={cn(
+          'absolute top-16 flex w-full flex-col overflow-hidden bg-primary transition-all duration-150 ease-in',
+          menuStyles,
+        )}
+      >
+        {navigation.map((item) => (
+          <NavItem {...item} />
```

Comment:
Let's add an optional onClick prop to handle the mobile navbar closing when the user clicks on any of the links for this:

1. On your `navbar/nav-item/model.ts` add a attribute on the interface with the optional syntax (the question mark before the colon) `onClick?: () => void`
2. On your NavItem component file bind the onClick prop to the onClick attribute of the `Link`
3. Finally on line 45 of NavbarMobile add the onClick prop and pass the `handleOpenMenu` as you did with the hambuger button to open the navbar